config/app/com/microsoft/azure/iotsolutions/uiconfig/webservice/v1/controllers/PackageController.java

```diff
@@ -39,16 +42,27 @@ public PackageController(IStorage storage) {
     }
 
     /**
-     * Retrieve all previously uploaded packages
-     * @return {@link PackageListApiModel}
+     * This function can be used to get packages with and without parameters
+     * PackageType, ConfigType. Without the query params this will return all
+     * the packages.
      */
     @Authorize("ReadAll")
-    public CompletionStage<Result> getAllAsync() throws BaseException {
-        return storage.getAllPackagesAsync().thenApplyAsync(m -> ok(toJson(new PackageListApiModel(m))));
+    public CompletionStage<Result> getFilteredAsync(String packageType, String configType) throws BaseException,
+            BadRequestException, ExecutionException, InterruptedException {
+
+        if (packageType == null && configType == null) {
+            return storage.getAllPackagesAsync().thenApplyAsync(m -> ok(toJson(new PackageListApiModel(m))));
+        }
+
+        if (packageType == null) {
```

Comment:
it's ok for configType to be null but not package type?

Reply:
yes, for edge we do not have config type parameter